QuantLib/ql/methods/finitedifferences/finitedifferencemodel.hpp

```diff
@@ -97,6 +110,8 @@ namespace QuantLib {
 
             Time dt = (from-to)/steps, t = from;
             evolver_.setStep(dt);
+            last_array_ = a;
+            last_dt_ = dt;
```

Comment:
You're saving these two at the beginning of the loop. Shouldn't you do it at the next-to-last step instead? As it is now, in the European engine you're taking the theta as the difference between now and maturity.